layers/+lang/emacs-lisp/packages.el

```diff
@@ -175,6 +178,33 @@
       (spacemacs/set-leader-keys-for-major-mode 'emacs-lisp-mode
         "dm" 'spacemacs/macrostep-transient-state/body))))
 
+(defun emacs-lisp/init-nameless ()
+  (use-package nameless
+    :defer t
+    :init (progn
+            (when (bound-and-true-p nameless-auto-mode)
```

Comment:
The standard "Spacemacs" way to do this would be to add a variable to `config.el` which would be called `emacs-lisp-nameless-mode` (or something similar). That variable would be set to nil and a user could enable it with     
```elisp
dotspacemacs-configuration-layers
'((emacs-lisp :variables emacs-lisp-nameless-mode t))
```
you could then check the value of it in `:init` and add the hook if necessary. The documentation would need to be updated to reflect this.